JASP-Engine/JASP/R/bainttestbayesianindependentsamples.R

```diff
@@ -24,335 +24,357 @@ BainTTestBayesianIndependentSamples <- function(jaspResults, dataset, options, .
 }
 
-.bainIndependentSamplesResultsTable <- function(dataset, options, jaspResults, missingValuesIndicator, ready, position) {
-
-  if (!is.null(jaspResults[["bainTable"]])) return() #The options for this table didn't change so we don't need to rebuild it
-
-  bainTable <- createJaspTable("Bain Independent Samples Welch's T-Test")
-  bainTable$dependOn(options =c("variables", "hypothesis", "bayesFactorType", "groupingVariable", "seed"))
-  bainTable$position <- position
-
-  bf.type <- options$bayesFactorType
-  BFH1H0 <- FALSE
-  bf.title <- "BF"
-
-  if (options$hypothesis == "allTypes") {
-          bainTable$addColumnInfo(name="Variable", type="string", title="")
-          bainTable$addColumnInfo(name = "type[equal]", type = "string", title = "Hypothesis")
-          bainTable$addColumnInfo(name="BF[equal]", type="number", title=bf.title)
-          bainTable$addColumnInfo(name="pmp[equal]", type="number", format="dp:3", title="Posterior probability")
-          bainTable$addColumnInfo(name = "type[greater]", type = "string", title = "Hypothesis")
-          bainTable$addColumnInfo(name="BF[greater]", type="number", title="bf.title")
-          bainTable$addColumnInfo(name="pmp[greater]", type="number", format="dp:3", title="Posterior probability")
-          bainTable$addColumnInfo(name = "type[less]", type = "string", title = "Hypothesis")
-          bainTable$addColumnInfo(name = "BF[less]", type = "number", title = bf.title)
-          bainTable$addColumnInfo(name="pmp[less]", type="number", format="dp:3", title="Posterior probability")
-  } else {
-          bainTable$addColumnInfo(name="Variable", type="string", title="")
-          bainTable$addColumnInfo(name = "hypothesis[type1]", type = "string", title = "Hypothesis")
-          bainTable$addColumnInfo(name="BF[type1]", type="number", title=bf.title)
-          bainTable$addColumnInfo(name="pmp[type1]", type="number", format="dp:3", title="Posterior probability")
-          bainTable$addColumnInfo(name = "hypothesis[type2]", type = "string", title = "Hypothesis")
-          bainTable$addColumnInfo(name="BF[type2]", type="number", title=bf.title)
-          bainTable$addColumnInfo(name="pmp[type2]", type="number", format="dp:3", title="Posterior probability")
-  }
+.readDataBainTwoSample <- function(options, dataset) {
+
+	all.variables 									<- unlist(options$variables)
+	grouping   										<- options$groupingVariable
+	read.variables 									<- c(all.variables, grouping)
+	if (options[["groupingVariable"]] == "")
+		grouping <- NULL
+
+	if (is.null(dataset)) {
+						trydata                	<- .readDataSetToEnd(columns.as.numeric=all.variables)
+						missingValuesIndicator 	<- .unv(names(which(apply(trydata, 2, function(x) { any(is.na(x))} ))))
+            dataset 								<- .readDataSetToEnd(columns.as.numeric=all.variables, columns.as.factor=grouping, exclude.na.listwise=read.variables)
+    }
+	
+	.hasErrors(dataset, type="factorLevels",
```

Comment:
I think it's better to be in line with the way other analyses do this: error checking and reading data are separate steps. Which imo makes sense because `readDataBainTwoSample` does not convey `checkErrors`

Reply:
Implemented this now